apps/etl-func/src/adapters/functions/message-status-ingestion.ts

```diff
@@ -0,0 +1,22 @@
+import { MessageStatus, messageStatusSchema } from "@/domain/message-status.js";
+import { IngestMessageStatusUseCase } from "@/domain/use-cases/ingest-message-status.js";
+import { CosmosDBHandler } from "@azure/functions";
+
+const parseMessageStatus = (input: unknown): MessageStatus | undefined => {
+  const { data, success } = messageStatusSchema.safeParse(input);
+  return success ? data : undefined;
+};
+
+const messageStatusIngestionHandler =
+  (ingestUseCase: IngestMessageStatusUseCase): CosmosDBHandler =>
+  async (documents: unknown[]) => {
+    const parsedMessageStatusBatch = documents.map(parseMessageStatus);
+    const messageStatusBatch = parsedMessageStatusBatch.filter(
+      (status): status is MessageStatus => status !== undefined,
+    );
```

Comment:
Can't we remove the parseMessageStatus function and connect the filter with the map to have it like this?
```suggestion
    const messageStatusBatch = documents
  .map((document) => {
    const { data, success } = messageStatusSchema.safeParse(document);
    return success ? data : undefined;
  })
  .filter((status): status is MessageStatus => status !== undefined);

```